spipe/main.c

```diff
@@ -18,14 +18,20 @@
 
 #include "pushbits.h"
 
+struct accept_state {
```

Comment:
I think the spiped and spipe code paths are sufficiently different here that trying to make the patches look similar is pointless.  In spiped we store the pointer to conndone in the accept_state structure because it needs to be passed to every connection we establish; but in spipe we're creating a connection from main() and we can just provide the pointer right there without stashing it in a structure temporarily.

And casting an (int *) to a (void *) is just fine, so nothing to worry about there.